### PR TITLE
Feature/cvsb 7677

### PR DESCRIPTION
**Pull request by m-ujaffer**

### CODE REVIEW

File: src/main/java/uk/gov/dvsa/service/PDFGenerationService.java [hunk 1]
```diff
@@ -10,6 +10,8 @@
 import uk.gov.dvsa.logging.LoggingExecutor;
 
 import java.io.ByteArrayOutputStream;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
```
Comment (KHodderDVSA) [inline src/main/java/uk/gov/dvsa/service/PDFGenerationService.java]: These are for the "output to file" bit. Shouldn't commit it

File: src/main/java/uk/gov/dvsa/service/RequestParser.java [hunk 2]
```diff
@@ -45,14 +45,17 @@ public RequestParser() {
...
         return executor.timed(() -> parse(input), EventType.CERT_REQUEST_PARSING);
     }
 
     private Document parse(Map<String, Object> input) {
         try {
+            System.out.println("CHECK INPUT IN PARSE -> " + input.toString());
             String documentName = readDocumentName(input);
             Class<? extends Document> documentType = readDocumentType(documentName);
             String documentJson = readRequestBody(input);
+            System.out.println("CHECK DOCUMENT JSON -> " + documentJson);
```
Comment (KHodderDVSA) [inline src/main/java/uk/gov/dvsa/service/RequestParser.java]: Still needed?

File: src/main/resources/views/CommercialVehicles/RWT_DATA.hbs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
...uFhFAHCALBFTk2xHYFjiYfkEWtUbEHTmebdXhpw52t9ktu7DjGA3+CQqZDw9fBK7lPj+f70O2zUP/+r2GmA1QVNWQfJR5rJW/wNozMwRCvo36TsruCxy0DwkKmQw7Hpa1SELyLeh60gS7lV6y0N+PJREs69NpAw7VXhjIneIXrqwMelUjVn7/ycuAD3SgfD/wJDJEOTxL8fTEwasv55dSlx+HjwObRcq50cfi4Lh+zpmyb++1IsBZ1U7Jkahxj33ht2lPzSOqPezVOy8nvKhIEIh0qyN0dyDTOY2ekOQYFOhiR5sC/JGqLDuiegeBeFVszxU3sf7V3bzuKwzAYgP3LTbNpQVMo03N33/8td9hp6ZaeOczIjL97FCHFKDiOLc2mcECL329qUWaptWdMy7sPMV05elhg7izh60TLa81nmHXgjzRbwiHFTU/SqrFfSTf/6D+a7GzsMK++u6K1UyaYdVoYhqIDf6TZEg4WuKUAJx3dP4VljOHd/7uYaaCwmJHeVeA9YBjTHM3Rs5JAXThsKE/ljePYx7OSh4pxjX1HZ25updTHON4V9NhwIFdhHNfLvwLaMkCaDeFwwo3dLAwm2t9nBgx02Dr6dJzPYKWhN7ZBy/uf/wwd7HAtxltAPToN7iX8aRxp0T5uZbRJFF+kdMXFpoL3IQ/rtBhc3iU0odzbnBkNZn93oDFB3IjWPhkaCo4m4W4tz68dDenAH/Us1ZpC0NKlTZBFBT1ZmbVrZSvXCqAtA9RjePJTlEbPSmqz+TpyQ2LpwB/1IL/kVzc4NAJSas5PSMok8r+Cer6TbQQ0KYL8J5Xd0wulJu1W3OmGEP/XQafBqTX2y1vdocFy54OgUZFS07LF/wVFLv+s5HTgj3pMA/AQ8gdtsp6V1Cr1QulTCPnVDVZ+olh9jZLRSjK65hJAfHVD/QKJYvVFalxw2N/ykWFA6nWuNWnxWbgOzbKq1RKgwzY+REERBNGx9hmQe1QygOdb38MFk1JLSlzhf9CTiytuMLjQTpRqvRIjpEdDGw5yT3vqmxQh+l5hJxl8kHzaU99mjxGyLxyMNqJUN4sTTDiRSP1wqDQa1CaHXc7oYxixtw0WgKZY1R1+H99M5eMsr8zbPiK53mNb4azaiQ3pH+4vSLtZnTv7G8gAAAAASUVORK5CYII=" alt="DVSA Crown" id="logo"/>
```
Comment (KHodderDVSA) [inline src/main/resources/views/CommercialVehicles/RWT_DATA.hbs]: Well, I guess that's one way to do it, but do you then need the "background-image" url css? Just causes confusion for those who come after

File: src/main/java/uk/gov/dvsa/model/cvs/certificateData/Signature.java [hunk 4]
```diff
@@ -31,4 +33,8 @@ public Signature setImageData(String imageData) {
     public String getFormattedImageData() {
         return "data:image/" + this.getImageType() + ";base64," + this.getImageData();
     }
+    public Signature setFormattedImageData() {
```
[the remark this answers is not in the document]
Comment (m-ujaffer) [reply src/main/java/uk/gov/dvsa/model/cvs/certificateData/Signature.java]: I was using this for unit test. It helped in alignment of the signature. Removed it now.

File: src/main/java/uk/gov/dvsa/PdfGenerator.java [hunk 5]
```diff
@@ -59,7 +59,7 @@ public ApiGatewayResponse handleRequest(Map<String, Object> input, Context conte
 
                 eventLogger.logEventWithTraceInfo(EventType.CERT_SERVER_RECEIVE, requestTracingService.getCurrentTracingInformation());
             }
-
+            System.out.println("CHECK INPUT IN HANDLEREQUEST -> " + input.toString());
```
[the remark this answers is not in the document]
Comment (m-ujaffer) [reply src/main/java/uk/gov/dvsa/PdfGenerator.java]: removed

File: src/main/java/uk/gov/dvsa/model/cvs/certificateData/RwtCertificateData.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+    @JsonProperty("Defects")
+    private String[] defects;
+
+    @JsonProperty("IsTrailer")
+    private  boolean isTrailer;
+
+    private boolean hasDefects;
+
+    private boolean hasNoDefects;
```
[the remark this answers is not in the document]
Comment (m-ujaffer) [reply src/main/java/uk/gov/dvsa/model/cvs/certificateData/RwtCertificateData.java]: removed. only getter functions are used.

File: src/main/resources/views/CommercialVehicles/RWT_DATA.hbs (hunk 3 above)
Comment (m-ujaffer) [reply src/main/resources/views/CommercialVehicles/RWT_DATA.hbs]: updated css

**Comment by m-ujaffer** [comment]
Changes for CVSB-11006 pushed on the same branch as it is a minor change in the footer for RWT template.

File: src/main/java/uk/gov/dvsa/service/PDFGenerationService.java [hunk 7]
```diff
@@ -34,21 +36,22 @@ public PDFGenerationService(ITextRenderer renderer) {
...
         }
-
         // render the remaining HTML documents
         htmlContents.stream().skip(1).forEach(this::renderHtmlDocument);
-
         renderer.finishPDF();
+        try(OutputStream outputStream = new FileOutputStream("RwtPdfOutput.pdf")) {
+            outputPdf.writeTo(outputStream);
+        } catch (Exception e) {
+            System.out.println("BAD THINGS. DEFINITELY DON'T COMMIT THIS CHANGE");
```
Comment (KHodderDVSA) [inline src/main/java/uk/gov/dvsa/service/PDFGenerationService.java]: Sorry, thought this part was self explanatory

File: src/main/java/uk/gov/dvsa/service/PDFGenerationService.java (hunk 1 above)
Comment (KHodderDVSA) [reply src/main/java/uk/gov/dvsa/service/PDFGenerationService.java]: Don't just resolve this - remove it

File: src/main/resources/assets/stylesheets/rwt.hbs [hunk 8]
```diff
@@ -30,7 +30,7 @@ padding: 5px;
 .border-defects{
 border: 1px solid;
 padding: 1px 2px 2px 1px;
-height: 70px;
+min-height: 70px;
```
Comment (KHodderDVSA) [inline src/main/resources/assets/stylesheets/rwt.hbs]: will need max-height too if you want it to not push out your footer

File: src/main/resources/views/CommercialVehicles/RWT_DATA.hbs [hunk 9]
```diff
@@ -22,14 +22,14 @@
         </div>
     </li>
     <li> {{#if rwtData.hasDefects}}
-        <p>Within the parameters laid out in paragraph 2 and 3, above, I certify that for the reason(s) shown below the motor vehicle or trailer was not shown to meet the inspection requirements.</p>
-        <p id="defects">
-            <ul class="border-defects">
+        <p style="margin-bottom:5px">Within the parameters laid out in paragraph 2 and 3, above, I certify that for the reason(s) shown below the motor vehicle or trailer was not shown to meet the inspection requirements.</p>
```
Comment (KHodderDVSA) [inline src/main/resources/views/CommercialVehicles/RWT_DATA.hbs]: inline styles not recommended

File: src/main/resources/assets/stylesheets/rwt.hbs (hunk 8 above)
Comment (m-ujaffer) [reply src/main/resources/assets/stylesheets/rwt.hbs]: I want it to push the footer as it makes the content really ugly when the defects overflow. I tried that then chose min-height.

**Pull request merged**